aiohttp/web_request.py

```diff
@@ -494,6 +497,52 @@ def if_unmodified_since(self) -> Optional[datetime.datetime]:
         """
         return self._http_date(self.headers.get(hdrs.IF_UNMODIFIED_SINCE))
 
+    @staticmethod
+    def _etag_values(etag_header: str) -> Iterator[ETag]:
+        """Extract `ETag` objects from raw header."""
+        if etag_header == ETAG_ANY:
+            yield ETag(
+                is_weak=False,
+                value=ETAG_ANY,
+            )
+        else:
+            for raw_etag in etag_header.split(","):
```

Comment:
What if ETag contains comma? `"1,2"`

Actually the approach with LIST_QUOTED_ETAG_RE was not bad. You need just add `|(.)` at the end of the regexp. That group will be matched only if there is a garbage, so you can check it and raise an exception.
```
LIST_QUOTED_ETAG_RE = re.compile(fr"{_QUOTED_ETAG}(?:\s*,\s*|$)|(.)")
```

Reply:
Yeah, completely forgot about this case while thinking about optimizations.

Thanks for the smart solution!